### PR TITLE
Fix some shape bugs

### PR DESCRIPTION
**Pull request by betterenvi**

### CODE REVIEW

**Comment by nsfinkelstein** [comment]
Great! Thanks for taking the time to make a pull request. Good catch.

**Comment by nsfinkelstein** [comment]
@betterenvi Just for the record (and my own curiosity) can you provide examples of cases that don't work before your fix, and do work afterwards?

Thanks

**Comment by nsfinkelstein** [comment]
@betterenvi On second thought - the change to the predictions loss split is unnecessary, because even though the predictions will have different sizes based on the task and output size, the loss will always be batchsize x 1.  That said, writing it this way might make it clearer.  I'm curious if there's a particular use case that piece of code caused problems with.

**Comment by betterenvi** [comment]
@n-s-f 
For regression task, if `output_size` is always 1, everything is fine. But if `output_size` is greater than 1, the original code will be broken.

**Comment by nsfinkelstein** [comment]
@betterenvi Thanks - that makes sense for changing the size of the batch place holders.

As far as I understand, though, this commit (https://github.com/n-s-f/hierarchical-rnn/pull/1/commits/3c502c5f5500ba6f483d2f83d2629b887b415921) shouldn't change anything because loss will always be batch_size x 1. 

Did you find that that commit fixes a problem, and if so, how?

**Comment by betterenvi** [comment]
@n-s-f 
There are some regression tasks where `output_size` is greater than 1, and in this case, loss has shape [batch_size, output_size]. At last, the following code will not work
```Python
loss = tf.reduce_mean(mapped[:, :, 0])
predictions = mapped[:, :, 1:]
```

**Comment by nsfinkelstein** [comment]
Ah, I see the issue. This is because of the custom regression loss function here: https://github.com/n-s-f/hierarchical-rnn/blob/master/hmlstm/hmlstm_network.py#L60

An alternative fix would be to reduce mean up there to get loss down to `batch_size x 1`, but this works too.

Thanks again!
